### ISSUE TITLE
Allow admin to change the date format

### ISSUE THREAD

**Issue by shoulders** [webmin/webmin#2310]
| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Ubuntu Linux 22.04.5              |
|  Usermin version     |  2.102                             |
|  Virtualmin version  |  7.20.2 Pro                        |
|  Theme version       |  21.20.7                           |
|  Apache version      |  2.4.52                            |
|  Package updates     |  34 package updates are available  |

## Background

![0e958b6d4576723015f0f3e908746198805bd7fc_2_690x262](https://github.com/user-attachments/assets/08edd3f3-e256-410d-8753-3631ab7e6820)

I find American dates tricky to understand and British dates have similar format so can can get mixed up.

## Feature request

Allow an admin to set a specific date format such as `2024-11-11 11.00:00`

W3TC format is used across MySQL, MariaDB and loads of other places. Especially useful for programmers.

![0b34f454e211710103ef4c99845fe4c2cd7b6e1d](https://github.com/user-attachments/assets/81814dbc-8a4f-437d-a40a-02aab92a2410)

## Additional

These pages might be suitable and were suggested here: https://forum.virtualmin.com/t/where-can-i-change-the-date-format/130302/6

### `Webmin ⇾ Change Language and Theme` page for user-specific settings

![image](https://github.com/user-attachments/assets/8a3a5b52-a392-4dc9-bae2-7e2509d08a53)

### `Webmin ⇾ Webmin Configuration: Language and Locale` to make global changes across the entire system.

![image](https://github.com/user-attachments/assets/a3855ffa-0d5f-42d5-bdb3-fc5be6f8cda0)

**Comment by iliajie** [comment]
I think you just need to find the right locale for your needs. The closest might be `en_DK.UTF-8`...

**Comment by shoulders** [comment]
but this will change other things like language files, it is really only the date for me. Also might be useful for those Americans who want to use a normal date format 😄 

https://www.localeplanet.com/icu/en-DK/index.html

Personally I would be happy with a checkbox to override the local date format to W3TC format.

**P.S.** I appreciate it might not be a 5 minute fix.

**Comment by iliajie** [comment]
> but this will change other things like language files,

What other language files?

> Also might be useful for those Americans who want to use a normal date format 😄

What is a normal date format? US format is `M/d/YYYY h:m` .

**Comment by shoulders** [comment]
> What other language files?

files was the wrong word, maybe I should of said individual translations, Currency symbols, things that are translated.

Also if you implement auto language selection from the browser this would revert me back to UK english. I am sure I wrote a language selection based on browser language. I will be going back to that software shortly but it is PHP based.

> What is a normal date format? US format is `M/d/YYYY h:m` .

Yes, I can never understand this format, not logical

**Comment by iliajie** [comment]
> Yes, I can never understand this format, not logical

Yeah, I’m not a fan either. The best format for me is always `yyyy/m/d H:mm` (24-hour format, not AM/PM), especially for file listing and sorting. For display, though, it should be based on locale, and here I don't really care. If you set your locale to `en-DK`, it adopts the ISO 8601 date format, which should perfectly align with what you want.

We could add support for custom locales, but I’m pretty sure no one would actually use it...

**Comment by Vitusware** [comment]
|  SYSTEM INFORMATION||
|------------------------------|-------------------------------|
|  OS type and version  |    Ubuntu Linux 24.04.2     |
|  Virtualmin version     |     7.30.8             |
| Kernel and CPU | Linux 6.8.0-60-generic on x86_64|
| Processor information| Intel(R) Core(TM) i5-2400 CPU @ 3.10GHz, 4 cores |



If I may add something to this discussion:

I came upon this thread because I have a problem that the date format is completely off, for some reason it happened 
and searched for a solution.
*I think* is began with the last update of webmin and a theme update, yesterday/ this morning.
Almost positive about that but but I'm not 100% sure..

Been struggling all day to get it right,  but it won't and I'm all confused.

I've tried all options Usermin configuration Language and locale , same for webmin and it just won't work right.

Dates in filemanager are always off, for some reason.
I've set everything to Nederlands (Dutch)  and the date format should be DD/MM/YYYY  but it is MM/DD/YY
and I'm not sure why, even on the dashboard the date is off..
System date/time is perfect. day month year/
Recent logons however it totally off with MM/DD/YY hh:ss

![Image](https://github.com/user-attachments/assets/760f55ea-3fe3-4104-9f46-1c5c39d1fcd0)

Maybe something is wrong in the last Authentic theme update.

Getting all back to English and Locale to English (United States),  login out and it, refreshed etc.

The date is MM/DD/YYY  and I think  that is correct for the US

![Image](https://github.com/user-attachments/assets/bc2bcf0a-ae7c-4e2f-bfb9-9594193c29d8)

But I can't get the date to DD/MM/YYY as it is used in The Netherlands what ever I try to do.

Letting me change the date format would make this so easy to do.

Thank you.

**Comment by iliaross** [comment]
Hello there!

For testing purposes, I have set it to:

<img width="1140" alt="Image" src="https://github.com/user-attachments/assets/53264256-f8a6-4209-9d38-b6a30d226d73" />

And, recent logins got updated after clicking force-refresh button at the top right of the dashboard:

<img width="1150" alt="Image" src="https://github.com/user-attachments/assets/87f0ebe7-ba37-47bb-bc84-148d67cca929" />

Do these look correct to you?

**Comment by Vitusware** [comment]
> Hello there!
> 
> For testing purposes, I have set it to:
> 
> <img alt="Image" width="1140" src="https://private-user-images.githubusercontent.com/4426533/457608187-53264256-f8a6-4209-9d38-b6a30d226d73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA1NDU2MTMsIm5iZiI6MTc1MDU0NTMxMywicGF0aCI6Ii80NDI2NTMzLzQ1NzYwODE4Ny01MzI2NDI1Ni1mOGE2LTQyMDktOWQzOC1iNmEzMGQyMjZkNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MjFUMjIzNTEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjhlM2RmNmNmMzdjY2FlYTZkZjZkZjY5YjMyZWQzNDRiMTQ0ZjM2ZTY4YjI3NDI5MTgxN2RmYmY5ZWU0MjhmYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ZjzWKKelYxwoxjjbSDR2iLpdb8-jr2BOWdyJ9w1691M">

> Do these look correct to you?

Hello there!

Yes they do, and now I understand what and where this is going wrong:

![Image](https://github.com/user-attachments/assets/228895d7-5133-48b9-9fb5-84ff451d0959)

I din't look further than Nederlands?? erm.. No.. because why should there be another option for the Netherlands. ??



But, thanks to you,  now I know there is.. 

The first one (red), not even sure why it is there,, is wrong! and in my opinion should not even be there..
 _If you look at the global locale it is set to Nederlands (nl) .. 🤦‍♂️  (that should be correct but it is not.._


The second one (green) is the one that you did. and only that one is correct!
And YES now everything is is the right (for me that is) order, 🥳

Thank you again for the heads up and quick reply..

Now on to bed, I can sleep now- haha

**Comment by robekras** [comment]
The date format should not be made dependent by the locale setting. 

In German we have a "allowed" date formatting of the traditional dd.mm.yyyy and the ISO date format (yyyy-mm-dd).
The ISO date format is yyyy-mm-dd and not yyyy/mm/dd and not yyyy.mm.dd or anything else.
The ISO date (with the -) was introduced not to be confused with existing date formats like mm/dd/yyyy,  dd/mm/yyyy, mm/dd/yy yy/mm/dd or anything else.

**Comment by iliaross** [comment]
> The date format should not be made dependent by the locale setting.

Probably, yet there is no clean and simple fix to this.

> In German we have a "allowed" date formatting of the traditional dd.mm.yyyy and the ISO date format (yyyy-mm-dd).

What do you suggest as a solution?
